node-api/handlers/beacon/header.go

```diff
@@ -57,22 +61,36 @@ func (h *Handler) GetBlockHeaderByID(c handlers.Context) (any, error) {
 		c, h.Logger(),
 	)
 	if err != nil {
-		return nil, err
-	}
-	slot, err := utils.SlotFromBlockID(req.BlockID, h.backend)
-	if err != nil {
-		return nil, err
+		return nil, errors.Wrap(err, "failed to bind and validate request")
 	}
-	header, err := h.backend.BlockHeaderAtSlot(slot)
-	if err != nil {
-		return nil, err
+	var header *ctypes.BeaconBlockHeader
+	var root common.Root
+
+	switch req.BlockID {
+	case utils.StateIDGenesis, "0":
+		header = h.backend.GenesisBlockHeader()
+		root = h.backend.GenesisBlockRoot()
+	default:
+		slot, errInSlot := utils.SlotFromBlockID(req.BlockID, h.backend)
```

Comment:
Here handling the special genesis case when we query using "genesis" or slot number 0.